### PR TITLE
Move some Version model methods into rails scopes

### PR DESCRIPTION
**Pull request by colby-swandale**
Just moving some methods into Rails scopes. I think it makes the model more readable and helps the clear up domain logic.

### CODE REVIEW

**Comment by simi** [comment]
:thinking: I'm do not see actual value to do this only in one model. If rubygems.org prefers scopes, let's make it unified everywhere and if possible add rubocop check to enforce this.

**Comment by colby-swandale** [comment]
I was working on something else and saw this as a quick win. If someone else wants to make a PR that does this same change everywhere then great. Please no rubocop enforcement.

**Comment by simi** [comment]
> I was working on something else and saw this as a quick win.

@colby-swandale I do not see any win actually, and it is just your code preference. Next time I'll go thru this code, I'll see another code-style than I do prefer (and which is used in Rubygem model for example) and open PR to revert this. We can run in circles forever and just make git history hard to read.

> Please no rubocop enforcement.

I'm not big rubocop fan as well, so I prefer to change code-style only when working on code around which is not this case. My proposed rubocop enforcement was the only way for me how to make this change consistent.

TL;DR I think we can leave it as is and close this.

**Comment by deivid-rodriguez** [comment]
@colby-swandale Can you explain the rationale for doing this? I grepped for `scope :` and there isn't any usages of Rails scopes in the current code base, so this would break the currently preferred style.

**Comment by deivid-rodriguez** [comment]
Sorry, I just read the first post where you already explained this

> I think it makes the model more readable and helps the clear up domain logic.

I don't really mind one style or the other, but look equally readable to me. But I don't like don't this just for this single model, introducing an inconsistency that people might want to fix later to appease their preferred style.

**Comment by colby-swandale** [comment]
fine
